sdk/core/System.ClientModel/src/Pipeline/HttpClientPipelineTransport.Request.cs

```diff
@@ -137,6 +137,21 @@ internal static HttpRequestMessage BuildHttpRequestMessage(PipelineRequest reque
             return httpRequest;
         }
 
+        private static HttpContent? GetHttpRequestMessageContent(BinaryContent? content, CancellationToken cancellationToken)
+        {
+            if (content is MultipartFormDataBinaryContent multipartContent)
```

Comment:
If we use this approach, it will transfer the headers without the generator or a protocol-method caller needing to do anything for that.